src/modules/elevationProfile/components/panel/ElevationProfile.js

```diff
@@ -331,10 +332,32 @@ export class ElevationProfile extends MvuElement {
 					fill: true,
 					borderWidth: 4,
 					backgroundColor: (context) => {
-						return this._getBackground(context.chart, elevationData);
+						const selectedAttribute = this.getModel().selectedAttribute;
+						if (!currentColors[selectedAttribute]) {
+							currentColors[selectedAttribute] = {};
+						}
+						if (!currentColors[selectedAttribute].backgroundColor) {
+							if (context.chart.chartArea) {
+								currentColors[selectedAttribute].backgroundColor = this._getBackground(context.chart, elevationData, selectedAttribute);
+							} else {
+								return ElevationProfile.BACKGROUND_COLOR;
+							}
+						}
+						return currentColors[selectedAttribute].backgroundColor;
```

Comment:
Do we need these checks? Can we reduce this to `ElevationProfile.BACKGROUND_COLOR`

Reply:
hm - the background part is entered twice for every point (about 6000 times over long distances), but it doesn't 'do' anything, it just returns what is computed in _getBackground (normally ElevationProfile.BACKGROUND_COLOR or _getTextTypeGradient for surface) once.